awscli/customizations/s3/filegenerator.py

```diff
@@ -208,13 +207,19 @@ def list_files(self, path, dir_op):
                         for x in self.list_files(file_path, dir_op):
                             yield x
                     else:
-                        size, last_update = get_file_stat(file_path)
-                        last_update = self._validate_update_time(
-                            last_update, path)
-                        yield (
-                            file_path,
-                            {'Size': size, 'LastModified': last_update}
-                        )
+                        stats = self._safely_get_file_stats(path, file_name=name)
```

Comment:
I'm referring to these lines:

* [original](https://github.com/aws/aws-cli/pull/2333/files#diff-2d6153b2f8310981a59f57a017f7a2a3L213)
* [new](https://github.com/aws/aws-cli/pull/2333/files#diff-2d6153b2f8310981a59f57a017f7a2a3R221)